src/components/admin/grading/GradingDashboard.tsx

```diff
@@ -43,7 +46,37 @@ const GradingDashboard: React.FC = () => {
     },
   });
 
+  const [{ data: branches, loading: branchesLoading }] = useAxios({
+    method: "GET",
+    url: apiUrl(Service.REGISTRATION, "/branches"),
+    params: {
+      hexathon: hexathonId,
+    },
+  });
+
+  if (branchesLoading) {
+    return <LoadingScreen />
+  }
+
+  const gradingEnabled = branches?.map((branch: any) => branch.grading.enabled).includes(true);
+
```

Comment:
we could remove this axios call by passing `gradingEnabled`  as a state prop from the Link from `Navigation.tsx`, since you're making the same call there as well. this reduces the number of calls we'd have to make. so even if someone were to access this route directly, the state variable would be null and they would be shown the disabled message.